cmd/kubelet/kubelet.go

```diff
@@ -80,6 +80,12 @@ func main() {
 		die(err)
 	}
 
+	// re-parse the command-line flags on top of the returned configuration
+	// we layer flags over file-based and remote configuration to
+	// preserve backwards compatibility across binary upgrades
+	// see: https://docs.google.com/document/d/18-MsChpTkrMGCSqAQN9QGgWuuFoK90SznBbwVkfZryo/
```

Comment:
maybe add a short blurb to a contributor doc instead of linking to a google doc?

Reply:
or an issue is probably fine

Reply:
Done #56171